packages/core/database/lib/entity-manager/index.js

```diff
@@ -633,53 +702,242 @@ const createEntityManager = (db) => {
+          if (isAnyToMany(attribute)) {
+            select.push(orderColumnName);
+          }
+          if (isBidirectional(attribute) && isManyToAny(attribute)) {
+            select.push(inverseOrderColumnName);
           }
 
-          if (!isNull(data[attributeName])) {
-            const insert = toAssocs(data[attributeName]).map((data) => {
-              return {
+          // only delete relations
+          if (isNull(cleanRelationData.set)) {
+            await deleteRelations({ id, attribute, joinTable, db }, { relIdsToDelete: 'all' });
+          } else {
+            const isPartialUpdate = !has('set', cleanRelationData);
+            let relIdsToaddOrMove;
+
+            if (isPartialUpdate) {
+              if (isAnyToOne(attribute)) {
+                cleanRelationData.connect = cleanRelationData.connect.slice(-1);
+                cleanRelationData.disconnect = [];
+              }
+              relIdsToaddOrMove = toIds(cleanRelationData.connect);
+              const relIdsToDelete = toIds(
+                differenceWith(isEqual, cleanRelationData.disconnect, cleanRelationData.connect)
+              );
+
+              await deleteRelations({ id, attribute, joinTable, db }, { relIdsToDelete });
+
+              // add/move
+              let max;
+              const currentMovingRels = await this.createQueryBuilder(joinTable.name)
+                .select(select)
+                .where({
+                  [joinColumn.name]: id,
+                  [inverseJoinColumn.name]: { $in: relIdsToaddOrMove },
+                })
+                .where(joinTable.on || {})
+                .execute();
+              const currentMovingRelsMap = currentMovingRels.reduce(
+                (acc, rel) => Object.assign(acc, { [rel[inverseJoinColumn.name]]: rel }),
+                {}
+              );
+
+              if (isAnyToMany(attribute)) {
+                max = (
+                  await this.createQueryBuilder(joinTable.name)
+                    .max(orderColumnName)
+                    .where({ [joinColumn.name]: id })
+                    .where(joinTable.on || {})
+                    .first()
+                    .execute()
+                ).max;
+              }
+
+              const nonExistingRelsIds = difference(
+                relIdsToaddOrMove,
+                map(inverseJoinColumn.name, currentMovingRels)
+              );
+
+              const maxResults = await db
+                .getConnection()
+                .select(inverseJoinColumn.name)
+                .max(inverseOrderColumnName, { as: 'max' })
+                .whereIn(inverseJoinColumn.name, nonExistingRelsIds)
+                .where(joinTable.on || {})
+                .groupBy(inverseJoinColumn.name)
+                .from(joinTable.name);
+
+              const maxMap = maxResults.reduce(
+                (acc, res) => Object.assign(acc, { [res[inverseJoinColumn.name]]: res.max }),
+                {}
+              );
```

Comment:
I think here we should only execute this if join table has the inverse order column
```suggestion
            let maxMap = {}
            if ( isAnyToMany(attribute) && isBidirectional(attribute)) {  
              const maxResults = await db
                .getConnection()
                .select(inverseJoinColumn.name)
                .max(inverseOrderColumnName, { as: 'max' })
                .whereIn(inverseJoinColumn.name, nonExistingRelsIds)
                .where(joinTable.on || {})
                .groupBy(inverseJoinColumn.name)
                .from(joinTable.name);

              maxMap = maxResults.reduce(
                (acc, res) => Object.assign(acc, { [res[inverseJoinColumn.name]]: res.max }),
                {}
              );
         }
```

Reply:
Also, I think it would improve the readability if we change:
`isManyToAny(attribute) && isBidirectional(attribute)` -> `hasInverseOrderColumn()`
`isAnyToMany(attribute)`-> `hasOrderColumn()`